src/auth/azureAuth.ts

```diff
@@ -0,0 +1,246 @@
+            this.onSignInStatusChangeEmitter.fire(this.signInStatusValue);
+        }
+    }
+
+    public async getSession(): Promise<Errorable<AuthenticationSession>> {
+        await this.initializePromise;
+        if (this.signInStatusValue !== "SignedIn") {
+            return { succeeded: false, error: `Not signed in (${this.signInStatusValue}).` };
+        }
+
+        if (this.tenants.length === 0) {
+            return { succeeded: false, error: "No tenants found." };
+        }
+
+        let tenant: Tenant;
+        if (this.tenants.length > 1) {
+            const selectedTenant = await quickPickTenant(this.tenants);
+            if (!selectedTenant) {
+                return { succeeded: false, error: "No tenant selected." };
+            }
+
+            tenant = selectedTenant;
+        } else {
+            tenant = this.tenants[0];
+        }
+
+        // Get a session for a specific tenant.
+        return await getArmSession(tenant.id, { createIfNone: true });
+    }
+}
+
+const sessionProvider = new AzureSessionProvider();
+
+export function signIn(): Promise<void> {
+    return sessionProvider.signIn();
+}
+
+export function getSignInStatus(): SignInStatus {
+    return sessionProvider.signInStatus;
+}
+
+export function getSignInStatusChangeEvent() {
+    return sessionProvider.onSignInStatusChangeEmitter.event;
+}
+
+export function getAuthSession(): Promise<Errorable<AuthenticationSession>> {
+    return sessionProvider.getSession();
+}
+
+export function getEnvironment(): Environment {
+    return getConfiguredAzureEnv();
+}
+
+export function getCredential(): TokenCredential {
+    return {
+        getToken: async () => {
+            const session = await sessionProvider.getSession();
+            if (failed(session)) {
+                throw new Error("No Microsoft authentication session found.");
+            }
+
+            return { token: session.result.accessToken, expiresOnTimestamp: 0 };
+        },
+    };
+}
+
+export function getTokenInfo(session: AuthenticationSession): Errorable<TokenInfo> {
+    const tokenParts = session.accessToken.split(".");
+    if (tokenParts.length !== 3) {
+        return { succeeded: false, error: `Access token not a valid JWT: ${session.accessToken}` };
+    }
+
+    const body = tokenParts[1];
+    let jsonBody: string;
+    try {
+        jsonBody = Buffer.from(body, "base64").toString();
+    } catch (e) {
+        return { succeeded: false, error: `Failed to decode JWT token body: ${body}` };
+    }
+
+    const jwt = parseJson<Jwt>(jsonBody);
+    if (failed(jwt)) {
+        return jwt;
+    }
+
+    const tokenInfo: TokenInfo = {
+        token: session.accessToken,
+        expiry: new Date(jwt.result.exp * 1000),
+    };
+
+    return { succeeded: true, result: tokenInfo };
+}
+
+export function getDefaultScope(endpointUrl: string): string {
+    return endpointUrl.endsWith("/") ? `${endpointUrl}.default` : `${endpointUrl}/.default`;
```

Comment:
💡 I wonder if we add sample above this for what is the `enpointURL` sample will look like et. al.

Reply:
Agree - added.